### PR TITLE
Replacing implicit class with case class + implicit conversion method

### PR DESCRIPTION
**Pull request by wongkk25**

### CODE REVIEW

File: src/main/scala/elo/package.scala [hunk 1]
```diff
@@ -31,11 +31,7 @@ package object elo {
...
   // =========================================
-  // Type ALL the things!
+  // Implicit conversion functions
 
-  implicit class SlackUserId(userId: String) {
-    // test of whether an ID mentioned is a valid player
-    // slack user IDs are prefixed with "UD"
-    def isValid: Boolean = userId.contains("UD")
-  }
+  implicit def toSlackUserId(s: String): SlackUserId = SlackUserId(s)
```
Comment (bticknor) [inline src/main/scala/elo/package.scala]: Don't we need to update all the places where we do string interpolation with SlackUserIds?  Challenge message, leaderboard, etc

Comment (wongkk25) [reply src/main/scala/elo/package.scala]: We won't need to update everywhere, but you did remind me to add in a toString override on SlackUserId. That should do the trick to render the user id properly in strings.

File: src/main/scala/elo/Player.scala [hunk 2]
```diff
@@ -13,4 +13,6 @@ case class SlackUserId(userId: String) {
   // test of whether an ID mentioned is a valid player
   // slack user IDs are prefixed with "UD"
   def isValid: Boolean = userId.contains("UD")
+
+  override def toString: String = userId
```
Comment (bticknor) [inline src/main/scala/elo/Player.scala]: 👍

File: src/main/scala/elo/package.scala (hunk 1 above)
Comment (bticknor) [reply src/main/scala/elo/package.scala]: Yeah that's what I meant I guess.  Thanks!

**Pull request merged**